src/Adapter/Title/CommandHandler/DeleteTitleHandler.php

```diff
@@ -28,27 +28,22 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Title\CommandHandler;
 
-use Gender;
+use PrestaShop\PrestaShop\Adapter\Title\AbstractTitleHandler;
 use PrestaShop\PrestaShop\Core\Domain\Title\Command\DeleteTitleCommand;
 use PrestaShop\PrestaShop\Core\Domain\Title\CommandHandler\DeleteTitleHandlerInterface;
 use PrestaShop\PrestaShop\Core\Domain\Title\Exception\DeleteTitleException;
-use PrestaShop\PrestaShop\Core\Domain\Title\Exception\TitleNotFoundException;
 
 /**
  * Handles command that delete title
  */
-class DeleteTitleHandler implements DeleteTitleHandlerInterface
+class DeleteTitleHandler extends AbstractTitleHandler implements DeleteTitleHandlerInterface
 {
     /**
      * {@inheritdoc}
      */
     public function handle(DeleteTitleCommand $command): void
     {
-        $title = new Gender($command->getTitleId()->getValue());
-
-        if (0 >= $title->id) {
-            throw new TitleNotFoundException(sprintf('Unable to find title with id "%d" for deletion', $command->getTitleId()->getValue()));
-        }
+        $title = $this->titleRepository->get($command->getTitleId());
 
         if (!$title->delete()) {
             throw DeleteTitleException::createDeleteFailure($command->getTitleId());
```

Comment:
handle method should be just one line
```suggestion
            $this->titleRepository->delete($command->getTitleId());
```
in repository you can reuse get(TitleId $titleId) and implement new delete(TitleId $titleId) method reusing AbstractObjectModelRepository::deleteObjectModel

Reply:
Fixed